app/models/category.rb

```diff
@@ -1,4 +1,4 @@
 class Category < ApplicationRecord
-	has_many :sub_categories
+	has_many :sub_categories, :dependent => :destroy	
```

Comment:
please use hash new syntax 
```suggestion
	has_many :sub_categories, dependent: :destroy	
```